Web_Development/FrontEnd/HTML/html_conventions_practices.md

```diff
@@ -0,0 +1,408 @@
+    </body>
+    ```
+
+### Close All HTML Elements
+
+In HTML, Sometimes if you run without closing any tag, It won't give you a syntax error.
+
+Well, If the code/element is working without closing tag, then that doesn't make sense or a good practice to not to close the tags.
+
+-   Good Practice:
+
+    ```sh
+    <body>
+      <p>Girlscript Foundation</p>
+      <p>Winter of Contributing</p>
+    </body>
+    ```
+
+-   Bad Practice:
+
+    ```sh
+    <body>
+      <p>Girlscript Foundation
+      <p>Winter of Contributing
+    </body>
+    ```
+
+### Always Specify required Attributes
+
+Like Images,
+
+Always specify the alt attribute for images. This attribute is important if the image for some reason cannot be displayed.
+
+Also, always define the width and height of images.Because the browser can reserve space for the image before loading and sometimes image ratio differs.
+
+-   Good Practice:
+
+    ```sh
+    <img src="gwoc.gif" alt="GWOC" style="width:128px;height:128px"/>
+    ```
+
+-   Bad Practice:
+
+    ```sh
+    <img SRC="html5.gif">
+    ```
+
+### Spaces, Equal Signs and Blank Lines
+
+HTML allows spaces around equal signs. But space-less is easier to read and groups entities better together.
+
+-   Do not add blank lines, spaces, or indentations without a reason.
+-   For readability, add blank lines to separate large or logical code blocks.
+
+-   Good Practice:
+
+    ```sh
+    <body>
+      <img src="gwoc.gif" alt="GWOC" style="width:128px;height:128px"/>
+        <p>Admins</p>
+
+        <p>Supervisors</p>
+        <p>Mentors</p>
+    </body>
+    ```
+
+-   Bad Practice:
+
+    ```sh
+    <body>
+
+    <img src = "gwoc.gif" alt = "GWOC" style = "width:128px;height:128px">
+
+        <p>Admins</p>
+
+        <p>Supervisors</p>
+
+        <p>Mentors</p>
+
+
+    </body>
+    ```
+
+### List and Table in HTML
+
+-   Good Table Example:
+
+    ```sh
+    <table>
+      <tr>
+        <th>Name</th>
+        <th>Description</th>
+      </tr>
+      <tr>
+        <td>A</td>
+        <td>Description of A</td>
+      </tr>
+      <tr>
+        <td>B</td>
+        <td>Description of B</td>
+      </tr>
+    </table>
+    ```
+
+-   Good List Example
+
+    ```sh
+    <ul>
+      <li>Managers</li>
+      <li>Admins</li>
+      <li>Supervisors</li>
+      <li>Mentors</li>
+      <li>Contributors</li>
+    </ul>
+    ```
+
+### Never Skip
+
+As we seen above, A HTML page can be validated without writing or closing a tag.
+
+-   The <title> element is required in HTML.
+-   The contents of a page title is very important for search engine optimization (SEO)!
+-   The page title is used by search engine algorithms to decide the order when listing pages in search results.
+-   Omitting <body> tag can produce errors in older browsers.
+-   Omitting <html> tag and <body> can also crash DOM and XML software.
+-   If we omit <head> tag, Browsers will add all elements before <body>, to a default <head> element.
```

Comment:
HTML tags need to be inside code:

Example, this is a `<tag>`.

Reply:
Updated